File: Data Structures & Algorithms/two-integer-sum/submission-6.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> seenBefore = new  HashMap<Integer,Integer>();

        for(int i=0; i<nums.length; ++i){
            if(seenBefore.get(target-nums[i]) != null){
              return new int[]{seenBefore.get(target-nums[i]), i};
            }
            seenBefore.put(nums[i], i);
        }

        return new int[]{-1,-1};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L, R = 0, len(numbers)-1

        while L < R:
           sm = numbers[L] + numbers[R]
           if sm == target:
              return [L+1,R+1]
           elif sm > target:
              R -= 1
           else:
              L += 1

        return [-1,-1]